resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -267,17 +311,18 @@ private void onResponseAorAAAA(
 
         // We aked for A/AAAA but we got only CNAME.
         if (!cnames.isEmpty()) {
-            onResponseCNAME(question, envelope, cnames, false);
+            onResponseCNAME(question, envelope, cnames, false, promise);
         }
     }
 
-    private void onResponseCNAME(DnsQuestion question, AddressedEnvelope<DnsResponse, InetSocketAddress> envelope) {
-        onResponseCNAME(question, envelope, buildAliasMap(envelope.content()), true);
+    private void onResponseCNAME(DnsQuestion question, AddressedEnvelope<DnsResponse, InetSocketAddress> envelope,
+                                 Promise<T> promise) {
+        onResponseCNAME(question, envelope, buildAliasMap(envelope.content()), true, promise);
     }
 
     private void onResponseCNAME(
             DnsQuestion question, AddressedEnvelope<DnsResponse, InetSocketAddress> response,
-            Map<String, String> cnames, boolean trace) {
+            Map<String, String> cnames, boolean trace, Promise<T> promise) {
```

Comment:
@vietj maybe Im missing something but it seems like we could end up not notify the promise.